pkg/common/encoding/proto_wrapper_api/triple_wrapper.pb.go

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package proto_wrapper_api
+
+import (
+	reflect "reflect"
+	sync "sync"
+)
+
+import (
+	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
+
+	protoimpl "google.golang.org/protobuf/runtime/protoimpl"
+)
+
+const (
+	// Verify that this generated code is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(20 - protoimpl.MinVersion)
+	// Verify that runtime/protoimpl is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(protoimpl.MaxVersion - 20)
+)
+
+type TripleRequestWrapper struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	// hessian4
+	// json
+	SerializeType string   `protobuf:"bytes,1,opt,name=serializeType,proto3" json:"serializeType,omitempty"`
+	Args          [][]byte `protobuf:"bytes,2,rep,name=args,proto3" json:"args,omitempty"`
+	ArgTypes      []string `protobuf:"bytes,3,rep,name=argTypes,proto3" json:"argTypes,omitempty"`
+}
+
+func (x *TripleRequestWrapper) Reset() {
+	*x = TripleRequestWrapper{}
+	if protoimpl.UnsafeEnabled {
+		mi := &file_triple_wrapper_proto_msgTypes[0]
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		ms.StoreMessageInfo(mi)
+	}
+}
+
+func (x *TripleRequestWrapper) String() string {
+	return protoimpl.X.MessageStringOf(x)
+}
+
+func (*TripleRequestWrapper) ProtoMessage() {}
+
+func (x *TripleRequestWrapper) ProtoReflect() protoreflect.Message {
+	mi := &file_triple_wrapper_proto_msgTypes[0]
+	if protoimpl.UnsafeEnabled && x != nil {
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		if ms.LoadMessageInfo() == nil {
+			ms.StoreMessageInfo(mi)
+		}
+		return ms
+	}
+	return mi.MessageOf(x)
+}
+
+// Deprecated: Use TripleRequestWrapper.ProtoReflect.Descriptor instead.
+func (*TripleRequestWrapper) Descriptor() ([]byte, []int) {
+	return file_triple_wrapper_proto_rawDescGZIP(), []int{0}
+}
+
+func (x *TripleRequestWrapper) GetSerializeType() string {
+	if x != nil {
+		return x.SerializeType
+	}
+	return ""
+}
+
+func (x *TripleRequestWrapper) GetArgs() [][]byte {
+	if x != nil {
+		return x.Args
+	}
+	return nil
+}
+
+func (x *TripleRequestWrapper) GetArgTypes() []string {
+	if x != nil {
```

Comment:
I think we don't need to test if `x` is nil or not, because it `x` is nil, calling a method on `nil` will cause panic.

Reply:
Btw, is this file generated by some tools?

Reply:
> Btw, is this file generated by some tools?

yes, its generated by `protoc` from file `triple_wrapper.proto`.


> I think we don't need to test if `x` is nil or not, because if `x` is nil, calling a method will cause panic.

I tested that calling a method from a nil pointer with an explicit context type will not cause a panic in golang, unless using its "member variables".